test/clj/rems/test/poller/email.clj

```diff
@@ -15,99 +15,105 @@
+              [{:to-user "decider",
+                :subject "Decision request",
+                :body "Dear decider,\nUser assistant has requested your decision on application 7.\nView the application: http://localhost:3001/#/application/7"}]
+              [{:to-user "handler",
+                :subject "New decision notification",
+                :body "Dear handler,\nUser decider has sent a decision on application 7.\nView the application: http://localhost:3001/#/application/7"}
+               {:to-user "assistant",
+                :subject "New decision notification",
+                :body "Dear assistant,\nUser decider has sent a decision on application 7.\nView the application: http://localhost:3001/#/application/7"}]
+              [{:to-user "applicant",
+                :subject "Your application has been approved",
+                :body "Dear applicant,\nYour application 7 has been approved.\nView your application: http://localhost:3001/#/application/7"}
+               {:to-user "somebody",
+                :subject "Your application has been approved",
+                :body "Dear somebody,\nYour application 7 has been approved.\nView your application: http://localhost:3001/#/application/7"}
+               {:to-user "member",
+                :subject "Your application has been approved",
+                :body "Dear member,\nYour application 7 has been approved.\nView your application: http://localhost:3001/#/application/7"}]
+              [{:to-user "applicant",
+                :subject "Your application has been closed",
+                :body "Dear applicant,\nYour application 7 has been closed.\nView your application: http://localhost:3001/#/application/7"}
+               {:to-user "somebody",
+                :subject "Your application has been closed",
+                :body "Dear somebody,\nYour application 7 has been closed.\nView your application: http://localhost:3001/#/application/7"}
+               {:to-user "member",
+                :subject "Your application has been closed",
+                :body "Dear member,\nYour application 7 has been closed.\nView your application: http://localhost:3001/#/application/7"}]]
+             (text/with-language :en
+               (fn [] (mapv #(#'rems.poller.email/event-to-emails-impl % application) events))))))
+    (let [reject-event {:application/id 7
+                        :event/type :application.event/rejected
+                        :event/actor "handler"}]
+      (is (= [{:subject "Your application has been rejected",
+               :body "Dear applicant,\nYour application 7 has been rejected.\nView your application: http://localhost:3001/#/application/7",
+               :to-user "applicant"}]
+             (text/with-language :en
+               (fn [] (#'rems.poller.email/event-to-emails-impl reject-event (dynamic/apply-events base [reject-event])))))))))
```

Comment:
I would extract a helper like this to avoid duplication in the tests:

```clj
(defn event-to-emails [events]
  (let [application (dynamic/apply-events nil events)]
    (text/with-language :en
      (fn [] (mapv #(#'rems.poller.email/event-to-emails-impl % application) events)))))
```

Usage:
```clj
(event-to-emails (concat base events))
```

Reply:
Or actually this helper is not adequate if you want to skip the already tested events in this second branch. Maybe this would work better:

```clj
(defn events-to-emails [old-events new-events]
  (let [application (dynamic/apply-events nil (concat old-events new-events))]
    (text/with-language :en
      (fn [] (mapv #(#'rems.poller.email/event-to-emails-impl % application) new-events)))))
```

Reply:
done